althea_kernel_interface/src/get_neighbors.rs

```diff
@@ -25,8 +26,8 @@ impl KernelInterface {
         Ok(vec)
     }
 
-    pub fn trigger_neighbor_disc(&self) -> Result<(), Error> {
-        for interface in self.get_interfaces()? {
+    pub fn trigger_neighbor_disc(&self, interfaces: &HashSet<String>) -> Result<(), Error> {
```

Comment:
this may be a decent change, I have no idea why we find neighs on all interfaces then discard the ones not on peer interfaces.

Reply:
Looks good to me, I just wanted to make it work 100% of the time so I just pinged everyone, but this should be more efficient. I wonder if there can be any edge case where pinging one interface gets you a response from another interface, but in that case the network is so broken that its not worth trying to probe